backend.native/tests/interop/objc/tests/tryRetainGC.m

```diff
@@ -0,0 +1,11 @@
+#import "tryRetainGC.h"
+#import <Foundation/NSArray.h>
+
+@implementation WeakRefHolder
+-(void)loadManyTimes {
+    NSMutableArray* array = [NSMutableArray new];
+    for (int i = 0; i < 10000; ++i) {
+        [array addObject:self.obj];
```

Comment:
So, during this `self.obj`, Obj-C runtime takes a lock (1); then calls `_tryRetain`; we intercept that and call `BackRefFromAssociatedObject::tryAddRef`; there we first increment refcount of Kotlin associated object; and then enqueue decrementing it back. And the problem is that this enqueuing could previously induce GC (because of a lot of queued decrements); and in `createGarbageNSObject` we've created a bunch of objects that happen to take the same lock (1) during their deallocation?

Reply:
Yes.